docs/extend/validation.md

```diff
@@ -0,0 +1,179 @@
+# Validation
+
+When users create or update data, you'll want to make sure it is valid; for instance, emails should follow a certain format, passwords have a minimum length, and the title field of discussions must be provided when starting a discussion.
+
+Flarum's validation system is just a wrapper around [Laravel Validation](https://laravel.com/docs/6.x/validation#manually-creating-validators). We recommend familiarizing yourself with the basics of that before proceeding.
+
+## Using a Validator
+
+Using a Flarum validator is very simple: all you need to do is inject an instance of the validator you need, and use its `assertValid` method on the data you are validating.
+
+Please note that the data you pass into a validator should be an associative array of keys to values, not an Eloquent model instance. For example:
+
+```php
+use Flarum\Group\GroupValidator;
+
+
+class SomeClass
+{
+    protected $validator;
+    public function __construct(GroupValidator $validator) {
+        $this->validator = $validator;
+    }
+
+    public function someMethod() {
+        $this->validator->assertValid($group->getDirty());
+    }
+}
+```
+
+If it fails validation, a `Illuminate\Validation\ValidationException` will be thrown with the validation errors, so that Flarum's error handling system can optimally present them to the user.
+
+Remember that you can turn an Eloquent instance into an associative array of attributes via:
+
+- The `$instance->getAttributes()` method if the instance hasn't been saved.
+- The `$instance->getDirty()` method if the instance has been saved.
```

Comment:
I'm not sure if that's correct or if I correctly understand what you wanted to describe.

- `getAttributes()` will get all attribute values as they are currently present on the model, saved or not
- `getDirty()` will get all attribute values that have been modified and not yet saved. It works for new or existing models
- `getChanges()` will get all attributes that were modified and are now saved
- `getOriginal()` will get all attribute values that were retrieved from the database

I think it would also be important here to mention that we commonly validate data before pushing it into the model. I think most extensions do that, and directly validate the data from the request.

Reply:
To be honest, this is more an instance of me not understanding part of Eloquent. Learned something new today!